### ISSUE TITLE
HTTP Config custom headers OR OpsGenie heartbeat support

### ISSUE THREAD

**Issue by therealgambo** [prometheus/prometheus#4314]
## Proposal
Trying to send a 'DeadManSwitch' alert to OpsGenie, however they expect a different `Authorization` header than what we can send via webhook

They expect `Authorization GenieKey <api-key>`, prometheus currently sends `Authorization Bearer <api-key>` when using the `bearer_token` setting in http_config

## Bug Report
**What did you do?**
Tried to configure DeadManSwitch / Heartbeat for OpsGenie using Webhooks + HTTP Config

**What did you expect to see?**
Webhook successs

**What did you see instead? Under which circumstances?**
HTTP 401 error because of headers issue as stated

**Environment**

* Prometheus version: `2.3.1`

* Alertmanager version: `0.15.0`

**Comment by brian-brazil** [comment]
You've got the wrong repository here, however we've discussed this in the past and do not plan on supporting OpsGenie heartbeats or similar as they give a false sense of security. An end-to-end test that goes from an alerting rule all the way through your paging provider to a deadmansswitch.

As a mater of general maintainability, we don't support bespoke or non-standard authentication methods either.

**Comment by therealgambo** [comment]
Do you mind sharing where exactly this was discussed in the past?

**Comment by brian-brazil** [comment]
I believe it was https://github.com/prometheus/alertmanager/pull/444

**Comment by juliusv** [comment]
Closing this issue here, as it's an Alertmanager issue/question and this is the Prometheus repository.

**Comment by dano-o** [comment]
I know this is a closed thread but wanted to point it out regardless that it is possible to ping OpsGenie from the AlertManager without custom headers.

1.) You can simply use the apiKey in the target URL:

https://api.opsgenie.com/v2/heartbeats/heartbeatname/ping?apiKey=XXXX


2.) You can use the Basic authentication method in AlertManager. OpsGenie will accept the following:

Header Key: Authorization
Header Value: basic base64(:$apiKey)

Basically, just leave the username empty, and use the apikey as the secret.

Hope this helps!

**Comment by therealgambo** [comment]
Thanks for this @dano-o. I can confirm your first solution worked, however when I tried the second option I could not get that working.

**Comment by VR6Pete** [comment]
Hi @dano-o - Are you setting this up as a Prometheus job in Prometheus.yml? How have you implemented this server side?
